deploy/osd-delete-backplane-script-resources/20-delete-backplane-script-resources.CronJob.yaml

```diff
@@ -0,0 +1,115 @@
+            nodeAffinity:
+              preferredDuringSchedulingIgnoredDuringExecution:
+              - preference:
+                  matchExpressions:
+                  - key: node-role.kubernetes.io/infra
+                    operator: Exists
+                weight: 1
+          tolerations:
+            - effect: NoSchedule
+              key: node-role.kubernetes.io/infra
+              operator: Exists
+          restartPolicy: Never
+          serviceAccountName: osd-delete-backplane-script-resources-sa
+          containers:
+            - name:  osd-delete-backplane-script-resources
+              imagePullPolicy: Always
+              image: image-registry.openshift-image-registry.svc:5000/openshift/cli:latest
+              args:
+                - /bin/bash
+                - -c
+                - |
+                  set -euxo pipefail
+
+                  NS="openshift-backplane-managed-scripts"
+                  LABEL="managed.openshift.io/backplane-job-uuid"
+
+                  ### Gather data
+                  # Format: pod_name pod_status
+                  PODS=$(oc get pod -n $NS --selector $LABEL -o custom-columns=NAME:.metadata.name,PHASE:.status.phase --no-headers)
+
+                  # Format: pod_name(running)
+                  PODS_RUNNING=$(echo "$PODS" | { grep -i Running || test $? = 1; } | awk '{print $1}')
+
+                  # Format: pod_name(non-running)
+                  PODS_NOT_RUNNING=$(echo "$PODS" | { grep -v -i Running || test $? = 1; } | awk '{print $1}')
+
+                  # Format: sa_name
+                  SAS=$(oc get serviceaccount -n $NS --selector $LABEL -o custom-columns=NAME:.metadata.name --no-headers)
+
+                  # Format: role_name role_ns
+                  ROLES=$(oc get role -A --selector $LABEL -o custom-columns=NAME:.metadata.name,NS:.metadata.namespace --no-headers) 
+
+                  # Format: rolebinding_name rolebinding_ns
+                  ROLEBINDINGS=$(oc get rolebinding -A --selector $LABEL -o custom-columns=NAME:.metadata.name,NS:.metadata.namespace --no-headers)
+
+                  # Format: clusterrole_name
+                  CROLES=$(oc get clusterrole --selector $LABEL -o custom-columns=NAME:.metadata.name --no-headers)
+
+                  # Format: clusterrolebinding_name
+                  CROLEBINDINGS=$(oc get clusterrolebinding --selector $LABEL -o custom-columns=NAME:.metadata.name --no-headers)
+
+                  ## We don't want to delete resources related to a running pod.
+                  for pod in $PODS_RUNNING
+                  do
+                    SAS=$(echo "$SAS" | grep -v "$pod")
+                    ROLES=$(echo "$ROLES" | grep -v "$pod")
+                    ROLEBINDINGS=$(echo "$ROLEBINDINGS" | grep -v "$pod")
+                    CROLES=$(echo "$CROLES" | grep -v "$pod")
+                    CROLEBINDINGS=$(echo "$CROLEBINDINGS" | grep -v "$pod")
+                  done
+
+                  ## Delete non-running pods
```

Comment:
I think we should implement some grace period here, because in the worse case, a just finished pod can be deleted immediately.

One idea is to check against the creation time of the pod and if it say <= `cron period` or some other cutoff, then delete it

Reply:
This could be hard to do in bash :/